content/200-concepts/100-components/01-prisma-schema/07-referential-actions.mdx

```diff
@@ -0,0 +1,327 @@
+
+These include:
+
+- [`Cascade`](#cascade)
+- [`Restrict`](#restrict)
+- [`NoAction`](#noaction)
+- [`SetNull`](#setnull)
+- [`SetDefault`](#setdefault)
+
+To understand how referential actions can impact your data it is important to understand how foreign key constraints work.
+
+[Foreign keys](../../../guides/general-guides/database-workflows/foreign-keys#overview) are used to link fields between tables and represent [relations](./relations) in your database.
+
+The following example shows `User` and `Post` models with a [one-to-many](./relations#one-to-many-relations) relationship. The `Post` model has an `authorId` field which is mapped to a foreign key constraint on the database table.
+
+```prisma file=schema.prisma
+model Post {
+  id        Int     @id @default(autoincrement())
+  title     String
+  author    User    @relation(fields: [authorId], references: [id])
+  authorId  Int
+}
+
+model User {
+  id    Int         @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+The foreign key ensures that when saving rows into the `Post` table, any value in the `authorId` [scalar field](./data-model#scalar-fields) must be referencing an existing value in the `id` column of the `User` table. 
+
+When a [migration](../prisma-migrate) is run the following SQL is produced:
+
+```sql
+CREATE TABLE "Post" (
+    "id" INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+    "title" TEXT NOT NULL,
+    "authorId" INTEGER NOT NULL,
+    FOREIGN KEY ("authorId") REFERENCES "User" ("id") ON DELETE RESTRICT ON UPDATE CASCADE
+);
+
+CREATE TABLE "User" (
+    "id" INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT
+);
+```
+
+And the foreign key relation that was created can be seen here:
+
+```sql
+FOREIGN KEY ("authorId") REFERENCES "User" ("id") ON DELETE RESTRICT ON UPDATE CASCADE
+```
+
+### Current behavior 
+
+Prisma creates foreign keys for each relation between data models, with the following defaults: 
+
+- `ON DELETE CASCADE` - The users posts are deleted when the user is deleted.
+- `ON UPDATE CASCADE` - When a users `id` is updated, the corresponding `authorId` on the `Posts` table is updated too.
+
+When invoking the [`delete()`](../../../guides/database/advanced-database-tasks/cascading-deletes/postgresql#91-validating-restrict) or [`deleteAll()`](../prisma-client/crud#delete-all-records) methods using the Prisma Client, a runtime check is performed and the deletion of records prevented if they are referencing related objects. **This prevents cascade behavior**. 
+
+If you need to replicate the cascade behavior, you _can_ use [`$raw`](../prisma-client/raw-database-access) SQL queries to [delete multiple referenced records](../prisma-client/crud#deleting-all-data-with-raw-sql--truncate) (such as users blog posts) and the deletion will be successful, this is because the Prisma Client will **not** perform runtime checks.
```

Comment:
```suggestion
If you need to actually use the cascade behavior configured in the database, you _can_ use [`$raw`](../prisma-client/raw-database-access) SQL queries to [delete multiple referenced records](../prisma-client/crud#deleting-all-data-with-raw-sql--truncate) (such as users blog posts) and the deletion will be successful. This is because the Prisma Client will **not** perform runtime checks on raw queries.
```